### PR TITLE
feat: move firmware to modules

### PR DESCRIPTION
**Pull request by chmelevskij**
this fixes the disconnect issue

### CODE REVIEW

**Review by haslinghuis — APPROVED** [review]
Thanks. Tested on Linux.
_Fixes also issue when user connects, then pushes flash button, re-plugging FC and with auto-connect enabled it will connect if on firmware_flasher tab._

File: src/js/tabs/firmware_flasher.js
Comment (McGiverGim) [inline src/js/tabs/firmware_flasher.js:1256]: The alignment of the elements has been broken. We can keep them aligned o simply change it to something less fancy but less dependent of changes:
```javascript
firmware_flasher.FLASH_MESSAGE_TYPES = {
    NEUTRAL : 'NEUTRAL', 
    VALID : 'VALID',
    INVALID : 'INVALID',
    ACTION : 'ACTION',
};
```

**Review by McGiverGim — CHANGES_REQUESTED** [review]
A little change, for the rest is perfect!

One module newbie question: I see a lot of references inside the file to `TABS.firmware_flasher.xxx`. I know this works because we continue "supporting" the old way of doing things, but, can't the reference to TABS be removed to let the module be more independent? Or this must wait until we remove the old way?

**Comment by chmelevskij** [comment]
> One module newbie question: I see a lot of references inside the file to TABS.firmware_flasher.xxx. I know this works because we continue "supporting" the old way of doing things, but, can't the reference to TABS be removed to let the module be more independent? Or this must wait until we remove the old way?

I'm just keeping everything as it is to reduce the amount of changes. Once we shipped all the modules there will be second pass of eliminating those cases on tab basis.

**Comment by blckmn** [comment]
AUTOMERGE: (fail)
 - Github identifies PR as mergeable -> pass
 - PR is not in draft -> pass
 - Assigned to a milestone -> fail
 - cooling off period lapsed (9/06/2021 05:34:23) -> pass
 - commit count less or equal to three (1 commits) -> pass
 - 'Do not merge' label NOT found -> pass
 - 'Needs coordination with betaflight' label NOT found -> pass
 - At least one 'RN:' label found -> pass
 - 'Tested' label found -> fail
 - assigned to an approver (None) -> fail
 - approver count at least three (1 approvers) -> fail

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - Github identifies PR as mergeable -> pass
 - PR is not in draft -> pass
 - Assigned to a milestone -> fail
 - cooling off period lapsed (10/06/2021 00:45:03) -> pass
 - commit count less or equal to three (1 commits) -> pass
 - 'Do not merge' label NOT found -> pass
 - 'Needs coordination with betaflight' label NOT found -> pass
 - At least one 'RN:' label found -> pass
 - 'Tested' label found -> fail
 - assigned to an approver (None) -> fail
 - approver count at least three (1 approvers) -> fail

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - Github identifies PR as mergeable -> pass
 - PR is not in draft -> pass
 - Assigned to a milestone -> fail
 - cooling off period lapsed (10/06/2021 00:48:16) -> pass
 - commit count less or equal to three (1 commits) -> pass
 - 'Do not merge' label NOT found -> pass
 - 'Needs coordination with betaflight' label NOT found -> pass
 - At least one 'RN:' label found -> pass
 - 'Tested' label found -> fail
 - assigned to an approver (None) -> fail
 - approver count at least three (1 approvers) -> fail

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - Github identifies PR as mergeable -> PASS
 - PR is not in draft -> PASS
 - Assigned to a milestone -> FAIL
 - cooling off period lapsed (10/06/2021 00:52:12) -> PASS
 - commit count less or equal to three (1 commits) -> PASS
 - 'Don't merge' label NOT found -> PASS
 - 'Needs coordination with betaflight' label NOT found -> PASS
 - At least one 'RN:' label found -> PASS
 - 'Tested' label found -> FAIL
 - assigned to an approver (None) -> FAIL
 - approver count at least three (1 approvers) -> FAIL

File: src/js/tabs/firmware_flasher.js
Comment (limonspb) [inline src/js/tabs/firmware_flasher.js:3]: Just for me to understand, do we follow C coding style where the function opening bracket should be on the next line?
https://github.com/betaflight/betaflight/blob/master/docs/development/CodingStyle.md

File: src/js/tabs/firmware_flasher.js
Comment (limonspb) [inline src/js/tabs/firmware_flasher.js:1227]: Just for me to understand, do we follow C coding style where the function opening bracket should be on the next line?
https://github.com/betaflight/betaflight/blob/master/docs/development/CodingStyle.md

Comment (chmelevskij) [reply src/js/tabs/firmware_flasher.js:1227]: No

Comment (chmelevskij) [reply src/js/tabs/firmware_flasher.js:1227]: and if we do, that's something what should be included in linter. And everything would need to be rewritten.

Comment (limonspb) [reply src/js/tabs/firmware_flasher.js:1227]: I think that's what I been told to do by @mikeller some time ago.
I understand lots of code already has been written the other way, but there is some other stuff that's against coding style. It does not mean we need to rewrite it right now.

Comment (chmelevskij) [reply src/js/tabs/firmware_flasher.js:1227]: that style of braces etc is not that common in JS world. So don't really see why we need to follow it.

As for not doing it everywhere, it will just lead to what we have now. Bunch of different standards and so called lava pattern. If we'd decide to strictly stick to one, it would have to be 1 PR with all the changes and CI/CD rule to prevent other styles. Otherwise it will be 100 + 1 different way to write code.

Comment (haslinghuis) [reply src/js/tabs/firmware_flasher.js:1227]: We should stick with the opening bracket on the same line. (But that's just my opinion)

Comment (limonspb) [reply src/js/tabs/firmware_flasher.js:1227]: We should just agree on something :) But who is to give a final word?

Comment (haslinghuis) [reply src/js/tabs/firmware_flasher.js:1227]: I like the way it's done here as most of the code does it this way :smiley: So agree with @chmelevskij here.

**Review by Asizon — CHANGES_REQUESTED** [review]
Tested and now all works as expected. @chmelevskij Can you made mcgiver requested changes and align these lines?

**Comment by haslinghuis** [comment]
Please let's approve this, the alignment isn't part of this PR as it's already there. I will put in a PR to fix this after this is merged.

**Review by limonspb — APPROVED** [review]

**Comment by Asizon** [comment]
@haslinghuis itsbroken here, after remove "TAB" it get not aliggned.

**Review by Asizon — APPROVED** [review]

**Comment by Asizon** [comment]
Ok fixed in https://github.com/betaflight/betaflight-configurator/pull/2513

**Comment by haslinghuis** [comment]
:smiley: it's blocked until @McGiverGim approves as the requested changes does block merging although all other requirements has passed :+1:

**Review by McGiverGim — APPROVED** [review]
The alignment was broken at this PR, not before. But ok, approved, I hope it will be fixed in a later PR.

**Comment by blckmn** [comment]
AUTOMERGE: (PASS)
 - Github identifies PR as mergeable -> PASS
 - PR is not in draft -> PASS
 - Assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - 'Don't merge' label NOT found -> PASS
 - At least one 'RN:' label found -> PASS
 - 'Tested' label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> PASS

**Comment by haslinghuis** [comment]
Not sure what happened here but I started over.

**Pull request merged**
